llvm/include/llvm/IR/Intrinsics.td

```diff
@@ -2861,6 +2861,69 @@ def int_experimental_convergence_anchor
 def int_experimental_convergence_loop
   : DefaultAttrsIntrinsic<[llvm_token_ty], [], [IntrNoMem, IntrConvergent]>;
 
+//===------- GPU Intrinsics -----------------------------------------------===//
+
+class GPUIntrinsic<LLVMType ret_type, string name>
+  : DefaultAttrsIntrinsic<[ret_type],
+                          [],
+                          [NoUndef<RetIndex>, IntrNoMem, IntrSpeculatable]>,
+                          ClangBuiltin<name>;
```

Comment:
As I mentioned on the thread I do not think this should be the first solution to reach for.

For example as-is this immediately breaks running AMDGPUAttributor, which will now incorrectly not detect uses of the underlying intrinsics hidden behind these wrappers. These wrappers need to have the same semantics as an unknown external call (as in, this should not use DefaultAddrsIntrinsic)

And need a test showing that AMDGPUAttributor infers none of the implicit argument attributes if there is a call to any of these

Reply:
It doesn't break amdgpu attributor. The attributor will correctly completely ignore these intrinsics, as it should do, unless at some point we decide the amdgpu attributor should start working in terms of target agnostic intrinsics at which point it'll work fine. We don't need an extra test showing that the amdgpu attributor works correctly when shown unknown intrinsics, because if it doesn't have that property already it's already severely broken.

Reply:
https://godbolt.org/z/9Gv4jsW3n even the undecorated fake.use infers everything